### PR TITLE
nsqd: persisted data repeatedly consumed

### PR DESCRIPTION
**Pull request by mreiferson**
hi，
i have a question about nsqd persisted meta file update.
my nsqd version:
> nsqd v0.3.7 (built w/go1.6)

when nsqd start up, it load persisted data from disk, and then subscriber consumed the msg, but the nsqd not update the persist meta file, and then if nsqd crash when it restart, my subscriber will consume the same messages again. I know nsqd guarantees "messages are delivered at least onceAnchor link for: messages are delivered at least once"， but is this scene can improve？

### CODE REVIEW

**Comment by mreiferson** [comment]
RFR

File: nsqd/diskqueue.go [hunk 1]
```diff
@@ -569,15 +569,15 @@ func (d *diskQueue) handleReadError() {
...
+	var wcount int64
 	var r chan []byte
 
 	syncTicker := time.NewTicker(d.syncTimeout)
 
 	for {
 		// dont sync all the time :)
-		if count == d.syncEvery {
-			count = 0
+		if wcount == d.syncEvery {
```
Comment (jehiah) [inline nsqd/diskqueue.go]: don't we want `(wcount + rcount) == d.syncEvery` ? 

And more generally, do we need to distinguish between read and write counts, can we always use a single `count` var?

Comment (mreiferson) [inline nsqd/diskqueue.go]: > can we always use a single count var?

That should simplify things.

**Comment by mreiferson** [comment]
ready @jehiah

**Comment by jinhao** [comment]
@mreiferson  
can release a new version of nsq

I try to get the latest code from github, and rebuild the nsq
but the nsqd cannot commuicate with nsqlookup
it seems new nsqd use protocol 'V2', but nsqlookupd still use 'V1'
nsqlookupd log

    [nsqlookupd] 2016/04/11 10:01:01.732021 CLIENT(127.0.0.1:63929): desired protocol magic '  V1'
    [nsqlookupd] 2016/04/11 10:01:01.732123 ERROR: [127.0.0.1:63929] - E_BAD_BODY IDENTIFY    missing fields
    [nsqlookupd] 2016/04/11 10:01:01.732179 CLIENT(127.0.0.1:63929): closing
    [nsqlookupd] 2016/04/11 10:01:01.732192 ERROR: client(127.0.0.1:63929) - E_BAD_BODY IDENTIFY missing fields

**Comment by judwhite** [comment]
@jinhao That looks like you have `nsqd -lookupd-tcp-address` set to connect to the nsqlookupd HTTP port (4161). Can you try 4160 instead? If that's not the case can you paste your `nsqd` and `nsqlookupd` args?

**Pull request merged**